core/src/program/blocks/span_block.rs

```diff
@@ -76,11 +82,26 @@ impl Span {
     #[must_use]
     pub fn replicate(&self, num_copies: usize) -> Self {
         let own_ops = self.get_ops();
+        let own_decorators = &self.decorators;
         let mut ops = Vec::with_capacity(own_ops.len() * num_copies);
-        for _ in 0..num_copies {
+        let mut decorators = DecoratorList::new();
+
+        for i in 0..num_copies {
+            // replicate decorators of a span block
+            for decorator in own_decorators {
+                decorators.push((own_ops.len() * i + decorator.0, decorator.1))
+            }
             ops.extend_from_slice(&own_ops);
         }
-        Self::new(ops)
+        Self::new(ops, decorators)
+    }
+
+    pub fn decorators(&self) -> &DecoratorList {
+        &self.decorators
+    }
+
+    pub fn decorator_iter(&self) -> DecoratorIterator {
+        DecoratorIterator::new(&self.decorators)
```

Comment:
Let's add basic doc comments to these methods.

Reply:
Done